### PR TITLE
Allow binary files whose datatype has has been registered as sniffable

### PR DESCRIPTION
**Pull request by gregvonkuster**
to be sniffed upon upload.

### CODE REVIEW

File: tools/data_source/upload.py [hunk 1]
```diff
@@ -274,13 +274,18 @@ def add_file( dataset, registry, json_file, output_path ):
...
                         ext = parts[-1].strip().lower()
-                        if not Binary.is_ext_unsniffable(ext):
-                            file_err( 'The uploaded binary file contains inappropriate content', dataset, json_file )
-                            return
-                        elif Binary.is_ext_unsniffable(ext) and dataset.file_type != ext:
-                            err_msg = "You must manually set the 'File Format' to '%s' when uploading %s files." % ( ext.capitalize(), ext )
-                            file_err( err_msg, dataset, json_file )
-                            return
+                        if not Binary.is_ext_unsniffable( ext ):
+                            format_information = Binary.is_sniffable_binary( dataset.path )
```
Comment (nsoranzo) [inline tools/data_source/upload.py:278]: `is_sniffable_binary()` seems to be called already at line 133, why is needed again here?

Comment (gregvonkuster) [inline tools/data_source/upload.py:278]: @nsoranzo This is a good question, but I'm not sure of the answer.  I wasn't debugging the code that early in the flow, so I missed that.  Are binary files that have been registered as sniffable upload-able?  If so, I'm not quite sure how I was not able to upload one.  I'll close this PR if it was just something strange I was doing.

Comment (gregvonkuster) [inline tools/data_source/upload.py:278]: @nsoranzo I reverted the code and confirmed that you are correct - the original code worked, so it must have been a bug I introduced somewhere else.  Sorry about this...

Comment (nsoranzo) [inline tools/data_source/upload.py:278]: No problem, I'm glad it's working for you.

**Pull request closed without merging**